### PR TITLE
refactor: abstract context menu

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves TODO/duplicate code

## How this PR fixes it

The context menus from the transaction list and sidebar Safe list have been abstracted into one `ContextMenu` component.

## How to test it

- Expand a transaction and observe that the context menu is as before.
- Observe that the context menu in the sidebar Safe list is as before.

## Screenshots

image

image

### CODE REVIEW

File: src/components/transactions/TxDetails/TxData/Transfer/index.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ const TransferTxInfo = ({ txInfo, txStatus }: TransferTxInfoProps) => {
   return (
     <Box>
       <TransferTxInfoSummary txInfo={txInfo} txStatus={txStatus} />
-      <Box display="flex" alignItems="flex-end">
+      <Box display="flex" alignItems="center">
```
Comment (DiogoSoaress) [inline src/components/transactions/TxDetails/TxData/Transfer/index.tsx:34]: Why this change? I suggest leaving it as it was

**Review by DiogoSoaress — APPROVED** [review]
I suggest revert the "ellipsis" button. It looks weird to be centered

File: src/components/transactions/TxDetails/TxData/Transfer/index.tsx (hunk 1 above)
Comment (iamacook) [reply src/components/transactions/TxDetails/TxData/Transfer/index.tsx:34]: The button is otherwise misaligned:

image

Comment (iamacook) [reply src/components/transactions/TxDetails/TxData/Transfer/index.tsx:34]: It looks misaligned on my end without the styling adjustment. I'll revert my changes though and we can look at it separately.
